src/uct/ib/ud/base/ud_def.h

```diff
@@ -155,9 +155,8 @@ typedef struct uct_ud_comp_desc {
 
 typedef struct uct_ud_zcopy_desc {
     uct_ud_comp_desc_t      super;
-    const void              *payload;
-    uint32_t                lkey;
-    uint32_t                len;
+    uct_iov_t               iov[UCT_IB_MAX_IOV];
```

Comment:
we can save a smaller array in "zdesc" - need to save only "buffer" and "length" for each entry, UD will never support strided anyway. also, "length" coult be as small as uint16_t, as well as iovcnt.